lib/include/kframework/avm/teal/teal-driver.md

```diff
@@ -1948,46 +1949,172 @@ Stateful TEAL Operations
+     andBool GROUP_IDX <Int ({getTxnField(getCurrentTxn(), GroupIndex)}:>Int)
+
+  rule <k> loadGroupScratch(GROUP_IDX, I) => . ...</k>
+       <stack> XS => 0 : XS </stack>
+       <stacksize> S => S +Int 1 </stacksize>
+       <currentTx> TX_ID </currentTx>
+       <transaction>
+         <txID> TX_ID </txID>
+         <groupIdx> GROUP_IDX </groupIdx>
+         <txScratch> M </txScratch>
+         ...
+       </transaction>
+    requires 0 <=Int I andBool I <Int MAX_SCRATCH_SIZE
+     andBool notBool (I in_keys(M))
+     andBool S <Int MAX_STACK_DEPTH
+     andBool GROUP_IDX <Int ({getTxnField(getCurrentTxn(), GroupIndex)}:>Int)
 
   rule <k> loadGroupScratch(_, I) => panic(INVALID_SCRATCH_LOC) ... </k>
     requires I <Int 0 orBool I >=Int MAX_SCRATCH_SIZE
 
-  rule <k> loadGroupScratch(TXN_IDX, _) => panic(FUTURE_TXN) ... </k>
-    requires TXN_IDX >=Int {getTxnField(getCurrentTxn(), GroupIndex)}:>Int
+  rule <k> loadGroupScratch(GROUP_IDX, _) => panic(FUTURE_TXN) ... </k>
+    requires GROUP_IDX >=Int {getTxnField(getCurrentTxn(), GroupIndex)}:>Int
 
-  rule <k> loadGroupScratch(TXN_IDX, _) => panic(TXN_OUT_OF_BOUNDS) ... </k>
-    requires TXN_IDX <Int 0 orBool TXN_IDX >=Int {getGlobalField(GroupSize)}:>Int
+  rule <k> loadGroupScratch(GROUP_IDX, _) => panic(TXN_OUT_OF_BOUNDS) ... </k>
+    requires GROUP_IDX <Int 0 orBool GROUP_IDX >=Int {getGlobalField(GroupSize)}:>Int
   
   rule <k> loadGroupScratch(_, _) => panic(STACK_OVERFLOW) ... </k>
        <stacksize> S </stacksize>
     requires S >=Int MAX_STACK_DEPTH
 ```
 
+### Inner transaction opcodes
+
+```k
+  rule <k> itxn_begin => . ...</k>
+       <innerTransactions>
+         .List => 
+         ListItem(<transaction>
+           <txID> "" </txID>
+           <txHeader>
+             // TODO Fee is calculated dynamically
+             <fee>         0                                         </fee>
+             <sender>      getGlobalField(CurrentApplicationAddress) </sender>
+             <firstValid>  getTxnField(getCurrentTxn(), FirstValid)  </firstValid>
+             <lastValid>   getTxnField(getCurrentTxn(), FirstValid)  </lastValid>
+             <genesisHash> .Bytes                                    </genesisHash>
+             <txType>       "unknown"                                </txType>
+             <typeEnum>     0                                        </typeEnum>
+             <groupID>      Int2String(GROUP_ID +Int 1)              </groupID>
+             <groupIdx>     0                                        </groupIdx>
+             <genesisID>    .Bytes                                   </genesisID>
+             <lease>        .Bytes                                   </lease>
+             <note>         .Bytes                                   </note>
+             <rekeyTo>      getGlobalField(ZeroAddress)              </rekeyTo>
+           </txHeader>
+           <txnTypeSpecificFields>
+             .AssetTransferTxFieldsCell
```

Comment:
Why do we use `.AssetTransferTxFieldsCell` here specifically? Is the choice of which sort of empty fields to put here arbitrary? I don't really have a suggesting, just asking for a clarification.

Perhaps it would make sense to leave a comment here explaining the choice.

Reply:
Fixed now. I was able to use `.Bag` instead.